accelerators/nvidia.go

```diff
@@ -131,28 +130,30 @@ func (nm *nvidiaManager) Destroy() {
 func (nm *nvidiaManager) GetCollector(devicesCgroupPath string) (stats.Collector, error) {
 	nc := &nvidiaCollector{}
 
-	if !nm.devicesPresent {
-		return nc, nil
+	if !nm.devicesPresent || len(nm.nvidiaDevices) == 0 {
```

Comment:
I think len(nm.nvidiaDevices) will be zero if !nm.nvmlInitialized is false?  I think that is why this case was handled below in the previous version...